<launch>

<arg name="robot_name" default="lolo"/>
<arg name="utm_zone" default="34"/>
<arg name="utm_band" default="V"/>
<arg name="captain_ip" default="192.168.1.90" />

<param name="utm_zone" type="int" value="$(arg utm_zone)"/>
<param name="utm_band" type="string" value="$(arg utm_band)"/> 

<include file="$(find lolo_description)/launch/lolo_description.launch">
    <arg name="namespace" value="$(arg robot_name)"/>
</include>

<!-- <include file="$(find tf_lat_lon)/launch/odom_tf.launch" ns="$(arg robot_name)">
    <arg name="robot_name" value="$(arg robot_name)"/>
</include> -->

<!--
<node pkg="lolo_drivers" type="old_pose_converter.py" name="old_pose_converter" output="screen" ns="$(arg robot_name)">
    <param name="robot_name" value="$(arg robot_name)"/>
</node>
-->
<!-- <node pkg="captain_interface" type="interface" name="captain_interface" output="screen" ns="$(arg robot_name)"/> -->

<include file="$(find captain_interface)/launch/interface.launch" ns="$(arg robot_name)">
    <arg name="captain_ip" value="$(arg captain_ip)"/>
</include>


<node name="map2ned" pkg="tf" type="static_transform_publisher" args="0 0 0 1.571 0 3.1416 map world_ned 100"/>
<node name="utm2map" pkg="tf" type="static_transform_publisher" args="535529 4864397 0 0 0 0 1 utm map 100" /> 

<group ns="lolo">
<group ns="dr">
    <include file="$(find tf_lat_lon)/launch/tf_lat_lon.launch">
        <arg name="frame" value="$(arg robot_name)/base_link"/>
        <arg name="lat_lon_topic" value="lat_lon"/>
    </include>
</group>
</group>

<node name="ins_republish" type="relay" pkg="topic_tools" args="/ixblue_ins_driver/ix/ins /lolo/core/ins" output="screen"/>
<node name="ins_gps_republish" type="relay" pkg="topic_tools" args="/ixblue_ins_driver/standard/navsatfix /lolo/core/gps" output="screen"/>
<node name="ins_to_dr" type="ins_to_dr.py" pkg="lolo_drivers" output="screen"/>
<node name="dvl_to_altitude" type="dvl_to_altitude.py" pkg="lolo_drivers" output="screen"/>

    <include file= "$(find smarc_acomms_nm3)/launch/full_acomms_stack.launch">
        <arg name="robot_name" value="$(arg robot_name)"/>
        <arg name="driver_port" value="/dev/ttyUSB0"/>
    </include>

    <node name="acomms_dr_publish" type="publish_dr.py" pkg="smarc_acomms" output="screen" ns="lolo"/>

</launch>
